go/vt/vtgate/executor_scatter_stats.go

```diff
@@ -91,11 +110,11 @@ func (e *Executor) gatherScatterStats() (statsResults, error) {
 		route := routes[i]
 		resultItems[i] = &statsResultItem{
 			Query:                  plan.Original,
-			AvgTimePerQuery:        time.Duration(plan.ExecTime.Nanoseconds() / int64(plan.ExecCount)),
-			PercentTimeOfReads:     float64(100 * plan.ExecTime / readOnlyTime),
-			PercentTimeOfScatters:  float64(100 * plan.ExecTime / scatterExecTime),
-			PercentCountOfReads:    float64(100 * plan.ExecCount / readOnlyCount),
-			PercentCountOfScatters: float64(100 * plan.ExecCount / scatterCount),
+			AvgTimePerQuery:        time.Duration(divideOrReturnZero(plan.ExecTime.Nanoseconds(), plan.ExecCount)),
+			PercentTimeOfReads:     100 * divideOrReturnZero(plan.ExecTime, readOnlyTime),
```

Comment:
Except for AvgTimePerQuery, seems like all of these are float64. You could just convert the arguments to float64 prior to division instead of after, which would avoid the panic since float64 has a concept of infinity and not-a-number.

If you do that, then for AvgTimePerQuery, maybe just add a one if-statement to check for a null divisor inside this for-loop instead of a whole helper function?

Reply:
Thanks for the comment. Let me close this one and simply make another one-line change for **AvgTimePerQuery**